raiden-ts/src/path/epics.ts

```diff
@@ -1,48 +1,203 @@
-  map,
-  withLatestFrom,
-  timeout,
   debounceTime,
-  pluck,
+  delay,
   distinctUntilChanged,
+  filter,
+  first,
   groupBy,
-  switchMap,
+  map,
+  mergeMap,
+  pluck,
+  publishReplay,
   scan,
   startWith,
+  switchMap,
   tap,
-  delay,
+  timeout,
+  withLatestFrom,
 } from 'rxjs/operators';
 import { fromFetch } from 'rxjs/fetch';
-import { isActionOf, ActionType } from 'typesafe-actions';
-import { bigNumberify, BigNumber } from 'ethers/utils';
-import { Zero, Two } from 'ethers/constants';
+import { ActionType, isActionOf } from 'typesafe-actions';
+import { BigNumber, bigNumberify, hexDataLength, hexlify, toUtf8Bytes } from 'ethers/utils';
+import { Two, Zero } from 'ethers/constants';
 import { Event } from 'ethers/contract';
-import { isNil } from 'lodash';
+import { get, isNil } from 'lodash';
 
 import { RaidenAction } from '../actions';
 import { RaidenState } from '../state';
 import { RaidenEpicDeps } from '../types';
 import { getPresences$ } from '../transport/utils';
 import { messageGlobalSend } from '../messages/actions';
-import { PFSCapacityUpdate, MessageType } from '../messages/types';
-import { signMessage } from '../messages/utils';
+import { MessageType, PFSCapacityUpdate } from '../messages/types';
+import { MessageTypeId, signMessage } from '../messages/utils';
 import { channelDeposited } from '../channels/actions';
 import { ChannelState } from '../channels/state';
 import { channelAmounts } from '../channels/utils';
-import { Address, UInt, Int, decode } from '../utils/types';
-import { losslessStringify, losslessParse } from '../utils/data';
+import { Address, decode, Int, Signature, Signed, UInt } from '../utils/types';
+import { encode, losslessParse, losslessStringify } from '../utils/data';
 import { getEventsStream } from '../utils/ethers';
-import { pathFind, pathFound, pathFindFailed, pfsListUpdated } from './actions';
+import {
+  clearIOU,
+  pathFind,
+  pathFindFailed,
+  pathFound,
+  persistIOU,
+  pfsListUpdated,
+} from './actions';
 import { channelCanRoute, pfsInfo, pfsListInfo } from './utils';
-import { PathResults, Paths } from './types';
+import { IOU, LastIOUResults, PathResults, Paths, PFS } from './types';
+import { concat } from 'ethers/utils/bytes';
+import { Signer } from 'ethers';
+import { memoize } from 'lodash/fp';
+import { UserDeposit } from '../contracts/UserDeposit';
+import * as t from 'io-ts';
+
+/**
+ * Codec for PFS API returned error
+ */
+const PathError = t.readonly(
+  t.intersection([
+    t.type({
+      errors: t.string,
+      /* eslint-disable-next-line @typescript-eslint/camelcase */
+      error_code: t.number,
+    }),
+    t.partial({
+      /* eslint-disable-next-line @typescript-eslint/camelcase */
+      error_details: t.type({
+        from_: Address,
+        to: Address,
+        value: UInt(32),
+      }),
+    }),
+  ]),
+);
+interface PathError extends t.TypeOf<typeof PathError> {}
+
+type IOUPayload = {
+  serviceAddress: Address;
```

Comment:
Do we really need a new type for it? Isn't `serviceAddress` the same as `iou.receiver`? If so, you can replace the whole type by just the `Signed<IOU>` object

Reply:
You are right. I forgot about that. Thanks :)